cognite/client/_api/datapoints.py

```diff
@@ -517,6 +498,248 @@ def __init__(self, config: ClientConfig, api_version: str | None, cognite_client
+            and copying status codes (automatically taken care of when requested, ``include_status=True``). The only assumption is that
+            the time series are already created in the target project.
+
+                >>> from cognite.client.utils import MIN_TIMESTAMP_MS, MAX_TIMESTAMP_MS
+                >>> target_client = CogniteClient()
+                >>> to_copy = client.time_series.list(data_set_external_ids="my-use-case")
+                >>> queries = [DatapointsQuery(external_id=ts.external_id) for ts in to_copy]
+                >>> for dps_chunk in client.time_series.data(
+                ...     queries,  # may be several thousand time series...
+                ...     chunk_size_time_series=20,  # control memory usage by specifying how many to iterate at a time
+                ...     chunk_size_datapoints=100_000,
+                ...     include_status=True,
+                ...     ignore_bad_datapoints=False,
+                ...     start=MIN_TIMESTAMP_MS,
+                ...     end=MAX_TIMESTAMP_MS + 1,  # end is exclusive
+                ... ):
+                ...     target_client.time_series.data.insert_multiple(
+                ...         [{"external_id": dps.external_id, "datapoints": dps} for dps in dps_chunk]
+                ...     )
+
+        """
+        # To make efficient usage of the API, we don't want a chunk size like 10 to send a million API requests when we can
+        # get 10k/100k datapoints per request. Thus, we round up the given chunk size to the nearest integer multiple of 100k,
+        # then subdivide and yield client-side (we use the raw limit also when dealing with aggregates):
+        request_limit = self._DPS_LIMIT_RAW * math.ceil(chunk_size_datapoints / self._DPS_LIMIT_RAW)
+        if (
+            not is_finite(chunk_size_datapoints)
+            or chunk_size_datapoints != request_limit
+            and request_limit % chunk_size_datapoints
+        ):
+            raise ValueError(
+                "The 'chunk_size_datapoints' must be a positive integer that evenly divides 100k OR an integer multiple of 100k "
+                f"(to ensure efficient API usage), not {chunk_size_datapoints}."
+            )
+        chunk_fn = functools.partial(split_into_chunks, chunk_size=chunk_size_datapoints)
+
+        if not (chunk_size_time_series is None or is_positive(chunk_size_time_series)):
+            raise ValueError(
+                f"'chunk_size_time_series' must be a positive integer or None, not {chunk_size_time_series}"
+            )
+
+        user_queries = [queries] if (is_single := isinstance(queries, DatapointsQuery)) else queries
+        dps_lst_cls: type[DatapointsArrayList | DatapointsList] = (
+            DatapointsArrayList if return_arrays else DatapointsList
+        )
+
+        # Note: Next major we should remove id/xid/instance_id from the other retrieve methods and just accept queries:
+        qs_per_id_type = defaultdict(list)
```

Comment:
Just to help readability
```suggestion
        qs_per_id_type: dict[str, list[DatapointsQuery]] = defaultdict(list)
```

Should this loop be a private classmethod of `DatapointsQuery`?